filesmanager/filesmanager.py

```
@@ -320,55 +356,127 @@ def get_content(self, data, suffix=''):
+            Each request must contain the following parameters:
+            - contents: the content to be added with the following format:
             {
-                "name": directory_name,
-                "type": "directory",
-                "path": f"{path}/{directory_name}" if path else directory_name,
-                "metadata": {},  # Empty for now but could be used to store the directory data needed by Chonky.
-                "children": [],
+                "rootFolderId": ...,
+                "treeFolders": {
+                    "id": "<rootFolderId>",
+                    "name": "Root",
+                    "type": "directory",
+                    "path": "Root",
+                    "parentId": "",
+                    "metadata": {},
+                    "children": [
+                        {
+                            "id": <folderId>,
+                            "name": "Folder 1",
+                            "type": "directory",
+                            "path": "Root/Folder 1",
+                            "parentId": "<rootFolderId>",
+                            "children": [
+                                {
+                                    "id": <fileId>,
+                                    "parentId": "<folderId>",
+                                    "name": "File 1",
+                                    "type": "file",
+                                    "path": "Root/Folder 1/File 1",
+                                }
+                            ]
+                        },
+                    ]
             }
+
+            - file(s): file(s) to be uploaded, in case the content to be added contains files.
+        """
+        try:
+            self.initialize_directories()
+            self.temporary_save_upload_files(request.params.items())
+            contents = json.loads(request.params.get("contents", "{}"))
+            if not contents:
+                return Response(status=HTTPStatus.BAD_REQUEST)
```

Could we return a descriptive error message in the response?